analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -232,16 +234,16 @@ semantics switch
                 _ => null
             };
 
-        private static ProgramState ProcessOperation(SymbolicContext context)
+        private static IEnumerable<SymbolicContext> ProcessOperation(SymbolicContext context)
```

Comment:
I started with an array and then changed it. I was trying to avoid the array creation here, as it is invoked only once, only for iterating purposes.
Do you really want it?

Reply:
No. Its fine.